blobfuse/src/BlobStreamer.cpp

```diff
@@ -101,14 +101,14 @@ BlobBlock* BlobStreamer::GetBlock(const char* file_name, uint64_t offset, Stream
         if (download_size > MAX_BLOCK_SIZE_FOR_SINGLE_READ && 
             obj->GetSize() > 0) 
         {
-            if ((start_offset + download_size) > obj->GetSize()) {
-                download_size = obj->GetSize() - start_offset;
-                if (download_size == 0) {
+            if ((start_offset + download_size) > obj->GetSize()) {                
+                if (obj->GetSize() <= start_offset) {
                     obj->UnLock();
                     syslog(LOG_ERR, "Failed to download block of %s with offset %lu.  Errno : (Out of range).\n", file_name, start_offset);
                     errno = 416;
                     return NULL;
                 }
+                download_size = obj->GetSize() - start_offset;
```

Comment:
If download size becomes 0 post this calculation, then also we need to error out as done in original code.

Reply:
Added a condition `download_size ==0.`, then two cases to make `download size ` zero will give out-of-range errors

Reply:
if (download_size == 0 || obj->GetSize() <= start_offset) {
is in correct check as dowload-size is being calculated at line 111. We need to put similar condition check and return failure post line 111

Reply:
done